src/Numeric/Log.hs

```diff
@@ -421,20 +423,24 @@ instance (Precise a, RealFloat a, Ord a) => Real (Log a) where
 
 data Acc1 a = Acc1 {-# UNPACK #-} !Int64 !a
 
-instance (Precise a, RealFloat a) => Monoid (Log a) where
-  mempty  = Exp negInf
-  {-# INLINE mempty #-}
-  mappend = (+)
-  {-# INLINE mappend #-}
-  mconcat [] = 0
-  mconcat (Exp z:zs) = Exp $ case List.foldl' step1 (Acc1 0 z) zs of
+instance (Precise a, RealFloat a) => Semigroup (Log a) where
+  (<>) = (+)
+  {-# INLINE (<>) #-}
+  sconcat (Exp z :| zs) = Exp $ case List.foldl' step1 (Acc1 0 z) zs of
     Acc1 nm1 a
       | isInfinite a -> a
       | otherwise    -> a + log1p (List.foldl' (step2 a) 0 zs + fromIntegral nm1)
     where
       step1 (Acc1 n y) (Exp x) = Acc1 (n + 1) (max x y)
       step2 a r (Exp x) = r + expm1 (x - a)
-  {-# INLINE mconcat #-}
+  {-# INLINE sconcat #-}
+
+instance (Precise a, RealFloat a) => Monoid (Log a) where
+  mempty  = Exp negInf
+  {-# INLINE mempty #-}
+#if !(MIN_VERSION_base(4,11,0))
+  mappend = (<>)
+#endif
```

Comment:
This removes the manual `mconcat` implementation, no? How about:

```haskell
mconcat [] = 0
mconcat zs = sconcat zs
```

Reply:
Fair point. Done.